### PR TITLE
implement PayPal orders as stand-alone post type

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/96074

This ticket puts in place the code, and part of the changes, needed to handle Orders not as a (fragile) by-product of Attendees but as a stand-alone custom post type.
Long list:

1. move the management and representation of Orders to separate class
2. rewire methods to use the new class and its repository-ish methods
3. add support for Denied (from the merchant), Refunded and Pending orders and their handling in the tables and reports.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;should deleting an attendee replenish a ticket stock?&quot; | [Main.php:1166](https://github.com/moderntribe/event-tickets/blob/9dce079b85ccb89c2feb23358e1fa995f87dbd61/src/Tribe/Commerce/PayPal/Main.php#L1166)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Commerce/PayPal/Main.php [hunk 1]
```diff
@@ -361,7 +361,19 @@ public function register_types() {
 			'rewrite'         => false,
 			'capability_type' => 'post',
 			'has_archive'     => false,
-			'hierarchical'    => true,
+			'hierarchical'    => false,
+		);
+
+		$order_post_args = array(
+			'label'           => 'Orders',
```
Comment (ggwicz) [inline src/Tribe/Commerce/PayPal/Main.php]: `i18n`

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;should deleting an attendee replenish a ticket stock?&quot; | [Main.php:1168](https://github.com/moderntribe/event-tickets/blob/5e967ec23e388ec227e0c8d1c7ddaa011887e933/src/Tribe/Commerce/PayPal/Main.php#L1168)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;should deleting an attendee replenish a ticket stock?&quot; | [Main.php:1168](https://github.com/moderntribe/event-tickets/blob/22172f3f874a523c0c12126f8137344f5573bb30/src/Tribe/Commerce/PayPal/Main.php#L1168)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
This is now ready for code review again

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Main.php:613](https://github.com/moderntribe/event-tickets/blob/1b984e42e03c57915db57d664a291adce5253684/src/Tribe/Commerce/PayPal/Main.php#L613)
:x: | There must be a single space before the last parenthesis | [Main.php:613](https://github.com/moderntribe/event-tickets/blob/1b984e42e03c57915db57d664a291adce5253684/src/Tribe/Commerce/PayPal/Main.php#L613)
:warning: | Comment refers to a TODO task &quot;should deleting an attendee replenish a ticket stock?&quot; | [Main.php:1175](https://github.com/moderntribe/event-tickets/blob/1b984e42e03c57915db57d664a291adce5253684/src/Tribe/Commerce/PayPal/Main.php#L1175)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Commerce/PayPal/Stati.php [hunk 2]
```diff
@@ -49,11 +49,79 @@ class Tribe__Tickets__Commerce__PayPal__Stati {
...
+			self::$not_completed,
+			self::$pending,
+			self::$refunded,
+			self::$denied,
+		);
+
+		return in_array( $payment_status, $legit ) ? $payment_status : self::$undefined;
+	}
+
+	public static function register_order_stati() {
```
Comment (bordoni) [inline src/Tribe/Commerce/PayPal/Stati.php]: Docblock.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Main.php:613](https://github.com/moderntribe/event-tickets/blob/f931576c8347029db4f52dc31062b7e1b94c5212/src/Tribe/Commerce/PayPal/Main.php#L613)
:x: | There must be a single space before the last parenthesis | [Main.php:613](https://github.com/moderntribe/event-tickets/blob/f931576c8347029db4f52dc31062b7e1b94c5212/src/Tribe/Commerce/PayPal/Main.php#L613)
:warning: | Comment refers to a TODO task &quot;should deleting an attendee replenish a ticket stock?&quot; | [Main.php:1175](https://github.com/moderntribe/event-tickets/blob/f931576c8347029db4f52dc31062b7e1b94c5212/src/Tribe/Commerce/PayPal/Main.php#L1175)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
added missing doc block

**Pull request merged**